CI worker tests are broken for gated datasets: ConnectionError

CI worker tests are broken for gated datasets: https://github.com/huggingface/dataset-viewer/actions/runs/10319430999/job/28567795990
```
FAILED tests/job_runners/config/test_split_names.py::test_compute_split_names_from_streaming_response[gated-False-SplitNamesFromStreamingError-DatasetNotFoundError] - AssertionError: assert 'ConnectionError' == 'DatasetNotFoundError'
  
  - DatasetNotFoundError
  + ConnectionError
FAILED tests/job_runners/dataset/test_config_names.py::test_compute_splits_response[gated-False-ConfigNamesError-DatasetNotFoundError] - AssertionError: assert 'ConnectionError' == 'DatasetNotFoundError'
  
  - DatasetNotFoundError
  + ConnectionError
```

--- a/services/worker/tests/job_runners/config/test_split_names.py
+++ b/services/worker/tests/job_runners/config/test_split_names.py
@@ -173,7 +173,7 @@ def test_doesnotexist(
         # should we really test the following cases?
         # The assumption is that the dataset exists and is accessible with the token
         ("does_not_exist", False, "SplitNamesFromStreamingError", "DatasetNotFoundError"),
-        ("gated", False, "SplitNamesFromStreamingError", "ConnectionError"),  # See: huggingface/huggingface_hub#2457
+        ("gated", False, "SplitNamesFromStreamingError", "ConnectionError"),  # See: huggingface/datasets#7109
         ("private", False, "SplitNamesFromStreamingError", "DatasetNotFoundError"),
     ],
 )

--- a/services/worker/tests/job_runners/config/test_split_names.py
+++ b/services/worker/tests/job_runners/config/test_split_names.py
@@ -173,7 +173,7 @@ def test_doesnotexist(
         # should we really test the following cases?
         # The assumption is that the dataset exists and is accessible with the token
         ("does_not_exist", False, "SplitNamesFromStreamingError", "DatasetNotFoundError"),
-        ("gated", False, "SplitNamesFromStreamingError", "DatasetNotFoundError"),
+        ("gated", False, "SplitNamesFromStreamingError", "ConnectionError"),  # See: huggingface/huggingface_hub#2457
         ("private", False, "SplitNamesFromStreamingError", "DatasetNotFoundError"),
     ],
 )

--- a/services/worker/tests/job_runners/dataset/test_config_names.py
+++ b/services/worker/tests/job_runners/dataset/test_config_names.py
@@ -99,7 +99,7 @@ def test_compute_too_many_configs(
         # should we really test the following cases?
         # The assumption is that the dataset exists and is accessible with the token
         ("does_not_exist", False, "ConfigNamesError", "DatasetNotFoundError"),
-        ("gated", False, "ConfigNamesError", "ConnectionError"),  # See: huggingface/huggingface_hub#2457
+        ("gated", False, "ConfigNamesError", "ConnectionError"),  # See: huggingface/datasets#7109
         ("private", False, "ConfigNamesError", "DatasetNotFoundError"),
     ],
 )

--- a/services/worker/tests/job_runners/dataset/test_config_names.py
+++ b/services/worker/tests/job_runners/dataset/test_config_names.py
@@ -99,7 +99,7 @@ def test_compute_too_many_configs(
         # should we really test the following cases?
         # The assumption is that the dataset exists and is accessible with the token
         ("does_not_exist", False, "ConfigNamesError", "DatasetNotFoundError"),
-        ("gated", False, "ConfigNamesError", "DatasetNotFoundError"),
+        ("gated", False, "ConfigNamesError", "ConnectionError"),  # See: huggingface/huggingface_hub#2457
         ("private", False, "ConfigNamesError", "DatasetNotFoundError"),
     ],
 )

